src/screens/AddVocab/AddVocab.jsx

```diff
@@ -1,198 +1,315 @@
-  //   setAge(event.target.value);
-  // };
-
-  //make api call to get group
-  // function getGroups(name) {
-  //   console.log(name);
-  //   axios({
-  //     method: "GET",
-  //     url: serverAddress + "/api/groups",
-  //     headers: {
-  //       "Content-Type": "application/json",
-  //       Authorization: `${jwt}`,
-  //     },
-  //     params: {
-  //       languagePackage: name,
-  //     },
-  //   })
-  //     .then((response) => {
-  //       setGroups(response.data);
-  //       console.log(response.data);
-  //     })
-  //     .catch(function (error) {});
-  // }
-
-  //create dropdown items for language packages
-  let languagePackageDropdownItems;
-  //show if array of groups is 0, because .map is not working with empty array
-  if (packages != null) {
-    languagePackageDropdownItems = groups.map((p) => {
+      .catch((err) => {
+        showSnack("error", t("global.fetchError"));
+      });
+  }, [showSnack, t]);
+
+  const onClear = useCallback(() => {
+    setSelectedPackage(null);
+    setSelectedGroup(null);
+    setForeignWord("");
+    setDescription("");
+    setTranslations(null);
+  }, [setSelectedPackage, setSelectedGroup, setForeignWord, setTranslations]);
+
+  const openPackageModal = useCallback(() => {
+    setShowAddPackage(true);
+  }, []);
+
+  const openGroupModal = useCallback(() => {
+    setShowAddGroup(true);
+  }, []);
+
+  const closePackageModal = useCallback(() => {
+    setShowAddPackage(false);
+  }, []);
+
+  const closeGroupModal = useCallback(() => {
+    setShowAddGroup(false);
+  }, []);
+
+  const packageAdded = useCallback(
+    (newPackage) => {
+      setSelectedPackage({ value: newPackage.id, label: newPackage.name });
+      closePackageModal();
+      fetchPackages();
+      showSnack("success", t("screens.addVocab.savePackageSuccessMessage"));
+    },
+    [closePackageModal, fetchPackages, t, showSnack]
+  );
+
+  const groupAdded = useCallback(
+    (newGroup) => {
+      closeGroupModal();
+      fetchPackages();
+      setSelectedGroup({ value: newGroup.id, label: newGroup.name });
+      showSnack("success", t("screens.addVocab.saveGroupSuccessMessage"));
+    },
+    [closeGroupModal, fetchPackages, t, showSnack]
+  );
+
+  const onSubmit = useCallback(() => {
+    const submitTranslations = translations.map((elem) => {
       return {
-        value: p.name,
-        label: p.name,
+        name: elem,
       };
     });
-  } else {
-    languagePackageDropdownItems = [];
-  }
-
-  //create dropdown items for groups
-  let groupDropdownItems;
-  //show if array of groups is 0, because .map is not working with empty array
-  if (groups != null) {
-    groupDropdownItems = groups.map((p) => {
-      return {
-        value: p.name,
-        label: p.name,
-      };
+
+    createVocabulary(
+      selectedPackage.value,
+      selectedGroup.value,
+      {
+        name: foreignWord,
+        translations: submitTranslations,
+        description,
+      },
+      active
```

Comment:
For now there are two disabled states. So we need two switches. One for disable the whole vocab and one for moving the vocab into a special drawer called staged.